src/main/java/net/minecraftforge/logging/CrashReportAnalyser.java

```diff
@@ -0,0 +1,280 @@
+     * or the mixin class and its target(s))
+     * **/
+    public static String appendSuspectedMods() {
+        StringBuilder stringBuilder = new StringBuilder();
+        stringBuilder.append("Suspected Mod");
+        stringBuilder.append(suspectedMods.size() == 1 ? ": " : "s: ");
+        if(suspectedMods.isEmpty()){
+            stringBuilder.append("NONE");
+        }
+        else {
+            suspectedMods.forEach((iModInfo, position) -> {
+                stringBuilder
+                        .append("\n\t")
+                        .append(iModInfo.getDisplayName())
+                        .append(" (").append(iModInfo.getModId()).append("),")
+                        .append(" Version: ").append(iModInfo.getVersion())
+                        .append("\n\t\t")
+                        .append(position);
+
+                iModInfo.getOwningFile().getConfig().<String>getConfigElement("issueTrackerURL").ifPresent(issuesLink -> {
+                    stringBuilder
+                            .append("\n\t\t")
+                            .append("Please report this crash to the mod author(s): ")
+                            .append(issuesLink);
+                });
+                stringBuilder.append("\n");
+            });
+        }
+
+        return stringBuilder.toString();
+    }
+
+
+    public static Map<String, IModInfo> getBasePackageModCache() {
+        return basePackageModCache;
+    }
+
+    public static Map<IModInfo, String> getSuspectedMods(){
+        return suspectedMods;
+    }
+
+    /**
+     * Tries to analyse the given exception and uncategorized stacktrace of the crash report.
+     * It checks the stacktrace for any occurrence of the base package names cached in the basePackageModCache,
+     * including mixin classes.
+     * **/
+    public static void analyseCrashReport(Throwable throwable, StackTraceElement[] uncategorizedStackTrace) {
+        try {
+            scanThrowable(throwable);
+            scanStacktrace(uncategorizedStackTrace);
+        }
+        catch(Throwable t){
+            LOGGER.error("Failed to analyse crash report!", t);
+        }
+    }
+
+    /**
+     * Checks the stacktrace of the given throwable and all its children for any occurrence of the base package names cached in the basePackageModCache,
+     * including mixin classes.
+     * **/
+    private static void scanThrowable(Throwable throwable){
+        scanStacktrace(throwable.getStackTrace());
+
+        if(throwable.getCause() != null && throwable.getCause() != throwable){
+            scanThrowable(throwable.getCause());
+        }
+    }
+
+    /**
+     * Checks the given stacktrace for any occurrence of the base package names cached in the basePackageModCache,
+     * including mixin classes.
+     * **/
+    private static void scanStacktrace(StackTraceElement[] stackTrace){
+        for(StackTraceElement stackTraceElement : stackTrace){
```

Comment:
spaces after keywords and between brackets please
```suggestion
    private static void scanStacktrace(StackTraceElement[] stackTrace) {
        for (StackTraceElement stackTraceElement : stackTrace) {
```